readme.md

```diff
@@ -0,0 +1,149 @@
+
+
+![Book Store App.jpeg](readmeImg%2FBook%20Store%20App.jpeg)
+# Book Store App
+## 🔬 Project Overview
+
+This project is a Book Store application built using Java and Spring Boot. It is designed to manage a catalog of books, provide user authentication and authorization, and ensure secure access to various operations. The project supports CRUD operations for book management and includes a user-friendly interface for accessing the API documentation through Swagger UI.
+
+* * *
+
+## 🛠️ Technologies and Tools
+
+- **Spring Boot**: Core framework for building the application.
+- **Spring Data JPA**: Manages database interactions.
+- **Spring Security**: Handles authentication and authorization.
+- **Liquibase**: Manages database migrations.
+- **Maven**: Project management and dependency resolution.
+- **H2 Database**: In-memory database for testing (CI).
+- **MySQL**: Main database for the application.
+- **Swagger**: API documentation and testing tool.
+- **JUnit & Mockito**: For unit testing and mocking.
+- **GitHub Actions**: Continuous Integration (CI) tool.
+- **Checkstyle**: Ensures code style consistency.
+
+* * *
+
+## 💡Features
+
+- User Registration and Authentication
+- Book Browsing and Searching
+- Shopping Cart Management
+- Order Processing
+- User Profile Management
+- Admin Dashboard for Managing Books and Orders
+- Secure RESTful API Endpoints
+
+* * *
+
+## 🏗️️ Installation
+
+1. Clone the repository:
+    ```bash
+    git clone https://github.com/fedaktaras/book-store-app
+    cd online-bookstore
+    ```
+2. Configure the database: Create a new MySQL database:
+    ```sql
+    CREATE DATABASE online_book_store;
+    ```
+   Update the `application.properties` file with your MySQL database credentials:
+    ```properties
+    spring.datasource.url=jdbc:mysql://localhost:3306/online_book_store
+    spring.datasource.username=root
+    spring.datasource.password=yourpassword
+    spring.jpa.hibernate.ddl-auto=update
+    spring.jpa.show-sql=true
+    ```
+3. Install dependencies and build the project:
+    ```bash
+    mvn clean install
+    ```
+4. Run the application:
+    ```bash
+    mvn spring-boot:run
+    ```
+   The server will start on `http://localhost:8080`.
+
+### Using Docker
+
+1. Build the Docker image:
+    ```bash
+    docker-compose up
+    ```
+
+* * *
+
+## 🧑‍💻Usage
+
+- The app will be installed with a set of data.
+- Use Postman or Swagger UI to interact with the API endpoints.
+- Refer to the valid JSON files in the `try-me.txt` file (just copy/paste text to the proper endpoint).
```

Comment:
better to create a Postman collection, put all the request bodies into there and share a link to it here. It will be more organized, instead of simple `.txt` file, also I couldn't find it